ui/components/data-sources/BaseDataSourceDialog.js

```diff
@@ -235,6 +267,15 @@ class BaseDataSourceDialog extends Component {
                 </Modal.Body>
 
                 <Modal.Footer>
+                    {type === DataSourceType.Postgres && (
+                        <Button
+                            bsStyle="info"
+                            onClick={() => this.onTest()}
+                            disabled={submitButtonDisabled}
```

Comment:
I will keep this even though user doesn't have to actually type data source name to test connection.
If fields are empty, test function doesn't work properly (uses default like localhost:5432, user:aliok password:none). So, let's make sure UI asks all fields to be filled, like "submit" button. 

A better way would be to use the same validation rules, but not require name input to be filled. I don't think this is worth the effort.

Reply:
I'm happy with this